### PR TITLE
Bug fix: map to list in lines 170:173

### LINKED ISSUE #16: calculate-tin.py is giving an error when we used it for a bed file generated with s. cerevisiae gtf file
**Describe the bug**
calculate-tin.py is giving an error when we used it for a bed file generated with s. cerevisiae gtf file. Below is the error we get:

```
@ 2021-07-13 16:36:37: Get BAM file(s) ...
Total 1 BAM file(s):
	results/samples/RPL7_3/map_genome/RPL7_3.se.Aligned.sortedByCoord.out.bam
@ 2021-07-13 16:36:37: Processing results/samples/RPL7_3/map_genome/RPL7_3.se.Aligned.sortedByCoord.out.bam
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/tavymi75/zarp/config/my_run/.snakemake/conda/c6ae2829/lib/python3.9/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/home/tavymi75/zarp/config/my_run/.snakemake/conda/c6ae2829/lib/python3.9/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "/home/tavymi75/zarp/config/my_run/.snakemake/conda/c6ae2829/bin/calculate-tin.py", line 361, in gf
    coverage = genebody_coverage(
  File "/home/tavymi75/zarp/config/my_run/.snakemake/conda/c6ae2829/bin/calculate-tin.py", line 260, in genebody_coverage
    start = positions[0] - 1
IndexError: list index out of range
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/tavymi75/zarp/config/my_run/.snakemake/conda/c6ae2829/bin/calculate-tin.py", line 561, in <module>
    main()
  File "/home/tavymi75/zarp/config/my_run/.snakemake/conda/c6ae2829/bin/calculate-tin.py", line 541, in main
    pool.map(gf, conditions)
  File "/home/tavymi75/zarp/config/my_run/.snakemake/conda/c6ae2829/lib/python3.9/multiprocessing/pool.py", line 364, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/home/tavymi75/zarp/config/my_run/.snakemake/conda/c6ae2829/lib/python3.9/multiprocessing/pool.py", line 771, in get
    raise self._value
IndexError: list index out of range
```
See also discussion here:
https://git.scicore.unibas.ch/zavolan_group/pipelines/zarp/-/issues/179

Reporter: @meric466

### PR DESCRIPTION
**Pull request by fgypas**
## Description

Fixes #16 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation updated

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by AngryMaciek** [comment]
Please remove `/.DS_Store` from the repo, these are macOS system files.

**Comment by fgypas** [comment]
The documentation can be updated after the docker is available in biocontainers (because we don't know the tag it will get).

File: .test/calculate-tin/sample.tsv [hunk 1]
```diff
@@ -1,40 +1,40 @@
 transcript	sample_name
```
Comment (AngryMaciek) [inline .test/calculate-tin/sample.tsv:1]: This is the expected output, right?

All of the non-zero values changed(!) It seems that the methods is calculating now sth different then before... How do we know it is still correct?

File: requirements.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 bx-python>=0.8.10
 guppy3>=3.1.0
 matplotlib>=3.3.4
-pandas>=1.2.3
+pandas>=1.1
```
Comment (AngryMaciek) [inline requirements.txt]: But why was this necessary? There were no issues with `pip` installation previously (?)

File: setup.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 setup(
     name='tin-score-calculation',
-    version='0.4',
+    version='0.4.1',
```
Comment (AngryMaciek) [inline setup.py]: If the method is calculating something different now (regardless if it was correct then or now) I think we should bump up major ver. (?)

**Review by AngryMaciek — COMMENTED** [review]
Apart from my other comments:
* I assume that changes to the `.py` scripts (**apart from** the fix by @meric466 ) were introduced by `black` only, so I do not check them.
* https://github.com/zavolanlab/tin-score-calculation/pull/17#issuecomment-879880443

File: requirements.txt (hunk 2 above)
Comment (fgypas) [reply requirements.txt]: Yes, I run into this, but I was not upgrading pip. I will reverse this.

File: setup.py (hunk 3 above)
Comment (fgypas) [reply setup.py]: Ok, 0.5 then?

Comment (AngryMaciek) [reply setup.py]: OK `0.5.0` is fine for me, since we don't have `1.0.0` yet. This would be a minor version bump then.

Comment (fgypas) [reply setup.py]: OK, done.

File: requirements.txt (hunk 2 above)
Comment (fgypas) [reply requirements.txt]: OK, done. Works fine.

File: .test/calculate-tin/sample.tsv (hunk 1 above)
Comment (fgypas) [reply .test/calculate-tin/sample.tsv:1]: Well, I am not sure about this...

Comment (fgypas) [reply .test/calculate-tin/sample.tsv:1]: @mzavolan So the issue here is that the old version of the tool and the new version give us different TIN scores. Can this be explained by the changes you and @meric466 introduced today?

Comment (AngryMaciek) [reply .test/calculate-tin/sample.tsv:1]: We generally observe that the scores were lower previously and are higher now. Given that we operate on a dummy `bam` file with not very much reads alligned I would tend to say that I expect the transcript coverage to be rather poor and thus - TIN score low. Much more in the range 0-20 then 20+ ... So I am leaning towards saying that the previous scores made more sense.

Comment (meric466) [reply .test/calculate-tin/sample.tsv:1]: I will check some mouse results I generated a couple of months ago, and I will rerun them with the new script and see how they look.

Comment (AngryMaciek) [reply .test/calculate-tin/sample.tsv:1]: And it seems I was wrong:  
TIN measures uniformity of the coverage, not the coverage _per se_, therefore my argument above might not be valid...

**Comment by AngryMaciek** [comment]
@meric466
Please make sure all the tests in the _CI_ pass, only then I can merge.
Also, take a look at the review comments here: https://github.com/zavolanlab/tin-score-calculation/pull/17#pullrequestreview-706390992

**Comment by fgypas** [comment]
> @meric466
> Please make sure all the tests in the _CI_ pass, only then I can merge.
> Also, take a look at the review comments here: [#17 (review)](https://github.com/zavolanlab/tin-score-calculation/pull/17#pullrequestreview-706390992)

@AngryMaciek @meric466 I just fixed the formatting issue, so the tests now pass.

**Comment by AngryMaciek** [comment]
Thanks a lot! But https://github.com/zavolanlab/tin-score-calculation/pull/17#issuecomment-879880443 still stands.

**Comment by meric466** [comment]
> Thanks a lot! But [#17 (comment)](https://github.com/zavolanlab/tin-score-calculation/pull/17#issuecomment-879880443) still stands.

Thanks @fgypas , @AngryMaciek I removed the file, is it OK now?

File: .test/calculate-tin/sample.tsv (hunk 1 above)
Comment (AngryMaciek) [reply .test/calculate-tin/sample.tsv:1]: The problem was diagnosed and resolved by @meric466
